build_and_release.sh

```diff
@@ -18,7 +18,10 @@ platforms=(
   windows-arm64
 )
 
-if [[ $GITHUB_REF = refs/tags/* ]]; then
+if [ -n "$GH_RELEASE_TAG" ]; then
+  echo "invoking release tag override $GH_RELEASE_TAG"
+  tag="$GH_RELEASE_TAG"
+elif [[ $GITHUB_REF = refs/tags/* ]]; then
```

Comment:
@manue1 : Could you elaborate on this approach versus deriving the tag from the reference?  Is there an example you can share that showcases a concrete example and how others might benefit from this change?

Reply:
We have a use case for an upcoming CLI extension where the code resides in one repo and has the built binaries published to another separate repo. (see [workflow](https://github.com/github/ghes-manage/actions/runs/6187204180/job/16796543601))

Our current plan is to trigger the release manually by dispatching the workflow since it would follow the GHES release window. Instead of creating a release tag in the source repo we'd therefore solely create the tag in the destination repo and `GITHUB_REF` would therefore not be set when the workflow is run.

I'm also unsure if this is a use case that many others will benefit from. 🤔